File: aoc-2015/src/main/java/net/bbelovic/adventofcode/day1/Day1PuzzlePart1.java
package net.bbelovic.adventofcode.day1;

import net.bbelovic.adventofcode.Puzzle;

public class Day1PuzzlePart1 implements Puzzle<String, Long> {

    public Long solve(String input) {
        long result = 0L;
        char[] chars = input.toCharArray();
        for (char c: chars) {
            if (c == ')') {
                result = result - 1;
            } else if (c == '(') {
                result = result + 1;
            }
        }
        return result;
    }
}

File: aoc-2020/src/main/kotlin/net/bbelovic/adventofcode/year2020/day1/Day1PuzzlePart1.kt
package net.bbelovic.adventofcode.year2020.day1

import net.bbelovic.adventofcode.Puzzle

class Day1PuzzlePart1: Puzzle<List<String>, Long> {
    override fun solve(input: List<String>): Long {
        val actualResult = Long.MIN_VALUE
        for (each in input) {
            val otherPart = 2020 - each.toLong()
            if (otherPart.toString() in input) {
                return otherPart * each.toLong()
            }
        }
        return actualResult
    }
}
